### DISCUSSION TITLE
Solving battery stability with code?

### DISCUSSION THREAD

**Discussion by josehdx** [tttapa/Control-Surface#941] [Q&A] [answered]
Dear community

I have noticed that the project that I'm working on has some stability issues when i power the esp32 board with a battery. I have noticed that the pitch bending operation tends to send inacurate messages when is not in use (actually when the position is at idle, i mean on the center) the midi messages are constantly sent, which is good because i don't want the device to sleep in battery saving mode. But the problem is, even using high accuracy value for pitchbendsender function, the values are big enough make a pitch fluctuation in real time, meaning that instead con central position cc63 it can go one or 2 cc values below, for example. That said, i noticed that when i power the board via usb with a good quality charger, the fluctuation does not happen, so it is a power issue. The battery is new, but probably not the highest quality, i wonder if is possible to solve this with code, if so, could anyone give me a hint? Or is there any other method to improve electrical signal? Thanks

**Comment by tttapa** [comment, marked as answer]
You could try to [increase the filtering in software](https://github.com/tttapa/Control-Surface/blob/39182bbab79487236c11579db3acfa336813bf54/src/AH/Settings/Settings.hpp#L66), but fundamentally, this is a hardware issue, and should be addressed at the hardware level.

I would recommend asking for advice on https://forum.arduino.cc/, be sure to provide a clear description of your problem and setup, including a detailed schematic (also add a link to this discussion).

**Reply by josehdx** [reply]
Thank you so much, I set the value to 2.5 and it solved the issue!

**Reply by tttapa** [reply]
2.5 is not a valid value, it has to be an integer.
A value of 2.5 will be truncated to 2, which was the default value, so this change cannot have fixed your problem, something else did.

**Reply by josehdx** [reply]
I swear that i only changed the value. It was 2 first. I tried with 3 and f4 but after updating the code on the board, the cc values jumped from cc127 to a lower one, something like  cc 70 or similar . I will revert the filter value to 2 and make additional tests and post the results. Thanks

**Reply by josehdx** [reply]
Ultra weird. Maybe the board is damaged or something but now it works perfectly with a value of 2. As far as i know you haven't done tests with esp32-s2 or s3 series but they have been a little problematic when using Arduino ide or vscode, i am currently using vscode but previous versions did not let me upload code  only via Arduino ide. The native usb implementation is messing around in my opinion i need to specify what mode i want to use, but for now, it wi.orks as intended. Thanks

**Reply by josehdx** [reply]
Ok. I have noticed, that when i modify the pbpotentiometer <<pitchbendersender>> and if i dont touch the filtering value (and i mean not modifying anything on this file) the constant reading goes back, but if i modified both files (even setting their values to the originals) the issue is solved. Maybe i am doing something wrong in the main cpp file, but for now, one more time the issue looks solved. Thanks again
